apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import sinon from 'sinon';
+import {expect} from '../../util/reconfiguredChai';
+import {mount} from 'enzyme';
```

Comment:
Where possible, use `shallow` rather than `mount` to ensure you're testing components in isolation. See "Shallow" and "Full" here: https://github.com/enzymejs/enzyme/tree/master/docs/api for more details.

Reply:
shallow doesn't seem to work here because I need to wrap the component with the Provider since it's a connected component. Do we prefer to use mount in this case or do something else?

Reply:
Huh. Looks like I've been doing it wrong (exporting a non-connected version used only for testing).
This is definitely a non-blocking comment.

Looks like the suggested method is to use shallow with dive and a wrapping component:
Shallow with dive: https://github.com/enzymejs/enzyme/issues/1002
Wrapping component: https://github.com/enzymejs/enzyme/issues/2202